### PR TITLE
[Task][Fe] Create Secure Store Library

### LINKED ISSUE #18: [Task][Fe] Create Secure Store Library

### PR DESCRIPTION
**Pull request by jemian868**
### [Github Task Link]
[[Task][Fe] Create Secure Store Library](https://github.com/orgs/Youngs-Milling-Software-Devs/projects/15/views/1?pane=issue&itemId=57825523)

### [Commands]
`npm i`
`npm run start`

### [Pre-Conditions]

None

### [Expected Output]

None

### [Notes]

None

### [Screenshots]

None

### CODE REVIEW

File: src/lib/SecureStore.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import * as SecureStore from "expo-secure-store";
+
+export const setSecureValue = async (key: string, value: string) => {
```
Comment (alOletres) [inline src/lib/SecureStore.ts:3]:
```suggestion
type SecureKey = "accessToken" |  "refreshToken"
export const setSecureValue = async (key: SecureKey, value: string) => {
```

File: src/screens/public/login/index.tsx [hunk 2]
```diff
@@ -65,12 +62,22 @@ export const LoginScreen = () => {
...
+            icon="key-variant"
             mode="contained"
-            onPress={() => navigation.navigate(EPublicScreen.BOTTOM_NAVIGATOR)}
+            onPress={handleFormSubmit(handleSubmit)}
+            // onPress={() => navigation.navigate(EPublicScreen.BOTTOM_NAVIGATOR)}
             style={loginButton}>
             Login
           </Button>
+
+          <Button
```
Comment (alOletres) [inline src/screens/public/login/index.tsx:73]:
```suggestion
          delete
```

File: src/screens/public/login/index.tsx [hunk 3]
```diff
@@ -26,11 +26,8 @@ const {
 } = styles;
 
 export const LoginScreen = () => {
-  const { navigation } = useHooks();
-
-  const { control } = useForm<Credentials>({
-    defaultValues: {},
-  });
+  const { navigation, control, handleSubmit, handleFormSubmit, sample } =
```
Comment (alOletres) [inline src/screens/public/login/index.tsx:29]:
```suggestion
omit sample property
```

File: src/state/reducers/hrmis/loginSlice.ts [hunk 4]
```diff
@@ -28,7 +32,24 @@ const loginSlice = createSlice({
       })
 
       .addCase(loginAsync.fulfilled, (state, action) => {
-        console.log("fulfilled", state, action);
+        if (isPayload(action.payload)) {
+          const { username, password, ...data } = action.payload.data;
+
+          if (data.access && typeof data.access === "string") {
+            setSecureValue("access", data.access);
```
Comment (alOletres) [inline src/state/reducers/hrmis/loginSlice.ts:39]:
```suggestion
            setSecureValue("accessToken", data.access);
```

**Review by alOletres — CHANGES_REQUESTED** [review]

File: src/lib/SecureStore.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import * as SecureStore from "expo-secure-store";
+
+type SecureKey = "accessToken" | "refreshToken";
+
+export const setSecureValue = async (key: SecureKey, value: string) => {
+  await SecureStore.setItemAsync(key, value);
+};
+
+export const getSecureValue = async (key: string) => {
```
Comment (alOletres) [inline src/lib/SecureStore.ts:9]:
```suggestion
export const getSecureValue = async (key: SecureKey) => {
```

File: src/screens/public/login/hook.ts [hunk 6]
```diff
@@ -1,8 +1,44 @@
...
+  });
+
+  const handleSubmit = () => {
+    const { username, password } = getValues();
+
+    dispatch(loginAsync({ username, password }));
+  };
+
+  useEffect(() => {
+    console.log("reduxResponse", isSuccess, message);
```
Comment (alOletres) [inline src/screens/public/login/hook.ts:37]:
```suggestion
delete console.log
```

**Review by alOletres — CHANGES_REQUESTED** [review]

**Comment by LodiAniez** [comment]
haha ang commands

**Pull request merged**